File: WareHouseManagement Mobile/warehouse_mobile/lib/models/location.dart
class Location {
  final String id;
  final String name;
  final String rackId;
  final String zone;
  final int row;
  final int grid;
  final int capacity;
  final String? productId;

  Location({
    required this.id,
    required this.name,
    required this.rackId,
    required this.zone,
    required this.row,
    required this.grid,
    required this.capacity,
    this.productId,
  });

  bool get isOccupied => productId != null;

  factory Location.fromJson(Map<String, dynamic> json) {
    return Location(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      rackId: json['rackId'] ?? '',
      zone: json['zone'] ?? '',
      row: json['row'] ?? 0,
      grid: json['grid'] ?? 0,
      capacity: json['capacity'] ?? 0,
      productId: json['productId'],
    );
  }
}

File: WareHouseManagement/WareHouseManagement API/Domain/Entities/Location.cs
using Domain.BaseEntities;
namespace Domain.Entities;

public class Location:BaseEntity
{
    public int Row {  get; set; }
    public int Grid { get; set; }
    public string Zone { get; set; }
    public int Capacity { get; set; }
    public string XCoordinates { get; set; }
    public string YCoordinates { get; set; }
    public string ZCoordinates { get; set; }
    public string Description { get; set; }
    public Inventory Inventory { get; set; }

    public Guid? RackId { get; set; } 
    public int? RowNumber { get; set; }
    public int? GridNumber { get; set; }

    public Rack Rack { get; set; }

}
